layout/Query.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const _ = require('lodash');
+
+class ServiceConfiguration {
+  constructor() {
+    this._score = [];
+    this._filter = [];
+  }
+
+  filter(view) {
+    this._filter.push(view);
+  }
+
+  score(view) {
+    this._score.push(view);
+  }
+
+  render(vs) {
```

Comment:
would you consider providing an error to a developer who failed to override the `render` method in subclasses?